numba/tests/test_array_methods.py

```diff
@@ -714,8 +714,24 @@ def test_array_T(self):
     def test_array_copy(self):
         self.check_layout_dependent_func(array_copy)
 
+    def check_object_copy(self, pyfunc):
+        def check_obj(obj):
+            cfunc = njit((typeof(obj),))(pyfunc)
+            expected = pyfunc(obj)
+            got = cfunc(obj)
+            self.assertPreciseEqual(expected, got)
+
+        check_obj((1, 2, 3))
+        check_obj([1.0, 2.0, 3.0])
+        check_obj(6)
```

Comment:
I don't know if I got the idea of this test. You should be testing if `np.copy(a)` works when `a` is not a numpy array, no?